src/main.rs

```diff
@@ -40,7 +56,7 @@ fn main() -> io::Result<()> {
 
     let scale = prog.value_of("scale").unwrap();
     let scale: usize = match scale.parse() {
-                                Ok(size) => size,
+                                Ok(size) => check_enlarge_scale(size),
                                 Err(_) =>  {
```

Comment:
I prefer just halt the program if user input wrong scale argument. We don't need check_enlarge_scale, just write:
```rust
let scale: usize = match scale.parse() {
  Ok(size) if size > 0 && size <= MAX_ENLARGE_SCALE => size,
  _ => {
    // ...
  }
```
Also note that:
1. use `log::error` macro instead of `println!`.
2. I prefer not to indent the arm of match that deep. One level of indent is fine.

Reply:
I agree your second opinion and I am finding a way to programing rust in such a clean style.
However, I also prefer to use functions to improve readability of the code. Therefore I am thinking about write a function which include type parsing and range checking. For example:
```rust
fn arg_check_range(arg: &str, range: (usize, usize)) -> Result<usize, String> {
    let (min, max) = range;
    match arg.parse::<usize>() {
        Ok(n) if min <= n && n <= max => Ok(n),
        Err(_) => Err(String::from("Please select an integer as argument")),
        _ => Err(format!("Please select integer in range {} to {}", min , max)),
    }
}
```
Then we can handle the result by `unwrap_or_else` in main function.
Maybe we can import this utility in the next pr.

Reply:
You don't have to wait next PR. If you think adding a handling function is better, just add it.